### PR TITLE
Emlid Edge support

### PR DESCRIPTION
**Pull request by staroselskii**
Hello guys!

As you might've heard we've released a new board called [Edge](https://emlid.com/edge/).

This PR adds initial support for this board. Some things like UAVCAN support are going to be added as soon as #6398 gets reworked.

Thanks!

### CODE REVIEW

File: libraries/AP_HAL/board/linux.h [hunk 1]
```diff
@@ -332,6 +332,19 @@
 #define HAL_GPIO_C_LED_PIN        16
 #define HAL_GPIO_LED_ON           LOW
 #define HAL_GPIO_LED_OFF          HIGH
+#elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_EDGE
+#define HAL_BOARD_LOG_DIRECTORY "/var/APM/logs"
```
Comment (lucasdemarchi) [inline libraries/AP_HAL/board/linux.h]: we should stop doing this... /var/lib/ardupilot/logs would be the more FHS-compliant dir. Can you move to it?

File: libraries/AP_HAL/board/linux.h [hunk 2]
```diff
@@ -332,6 +332,19 @@
...
+#define HAL_BOARD_TERRAIN_DIRECTORY "/var/APM/terrain"
```
Comment (lucasdemarchi) [inline libraries/AP_HAL/board/linux.h]: ditto

File: libraries/AP_HAL/board/linux.h [hunk 3]
```diff
@@ -332,6 +332,19 @@
...
+#define HAL_INS_DEFAULT HAL_INS_EDGE
+#define HAL_INS_MPU60x0_NAME "mpu60x0"
+#define HAL_INS_MPU60x0_NAME_EXT "mpu60x0ext"
+#define HAL_BARO_DEFAULT HAL_BARO_MS5611_SPI
+#define HAL_BARO_MS5611_NAME "ms5611"
+#define HAL_BARO_MS5611_USE_TIMER false
```
Comment (lucasdemarchi) [inline libraries/AP_HAL/board/linux.h]: this can go away. nobody uses timer anymore

File: libraries/AP_HAL/board/linux.h [hunk 4]
```diff
@@ -332,6 +332,19 @@
...
+#define HAL_COMPASS_DEFAULT HAL_COMPASS_HIL
```
Comment (lucasdemarchi) [inline libraries/AP_HAL/board/linux.h]: is this correct??

File: libraries/AP_HAL/board/linux.h (hunk 1 above)
Comment (staroselskii) [reply libraries/AP_HAL/board/linux.h]: Yeah. Sure. Thanks!

File: libraries/AP_HAL/board/linux.h (hunk 3 above)
Comment (staroselskii) [reply libraries/AP_HAL/board/linux.h]: Thanks. Totally missed that.

File: libraries/AP_HAL_Linux/SPIDevice.cpp [hunk 5]
```diff
@@ -123,6 +123,12 @@ SPIDesc SPIDeviceManager::_device[] = {
     SPIDesc("aeroio", 1, 1, SPI_MODE_0, 8, SPI_CS_KERNEL,  10*MHZ, 10*MHZ),
     SPIDesc("bmi160", 3, 0, SPI_MODE_3, 8, SPI_CS_KERNEL, 1*MHZ, 10*MHZ),
 };
+#elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_EDGE
+SPIDesc SPIDeviceManager::_device[] = {
+    SPIDesc("mpu60x0",    0, 1, SPI_MODE_0, 8, SPI_CS_KERNEL,  1*MHZ, 20*MHZ),
+    SPIDesc("mpu60x0ext",    0, 2, SPI_MODE_0, 8, SPI_CS_KERNEL,  1*MHZ, 20*MHZ),
```
Comment (lucasdemarchi) [inline libraries/AP_HAL_Linux/SPIDevice.cpp]: Are you sure 20 is ok rather than 11 like the others? See aadc1643fca834487f38584e9268f2f947e7c2b7

Comment (staroselskii) [reply libraries/AP_HAL_Linux/SPIDevice.cpp]: Thanks, Lucas. I have seen some temperature-related FIFO resets. Thank you for noticing this. I don't see a reason to maximize the frequency in this case, so I'll just do the same thing as others.

But I'd really like to hear a little more how did you glue this reset to the SPI speed.

**Comment by dgrat** [comment]
Looks good to me

File: libraries/AP_HAL_Linux/SPIDevice.cpp (hunk 5 above)
Comment (lucasdemarchi) [reply libraries/AP_HAL_Linux/SPIDevice.cpp]: I think I noticed the resets right after rebasing and checking the IMU driver started monitoring the temperature. It was a long time ago, I don't remember exactly :(

**Comment by WickedShell** [comment]
I'm not sure if it should go on Travis or Semaphore but we should probably add this to one of the CI systems now.

**Pull request merged**